src/polysolve/LinearSolverCuSolverDN.cu

```diff
@@ -0,0 +1,150 @@
+#ifdef POLYSOLVE_WITH_CUSOLVER
+
+////////////////////////////////////////////////////////////////////////////////
+#include <polysolve/LinearSolverCuSolverDN.cuh>
+#include <fstream>
+#include <string>
+#include <iostream>
+
+#define gpuErrchk(ans){ gpuAssert((ans), __FILE__, __LINE__); }
+
+inline void gpuAssert(cudaError_t code, const char *file, int line,
+                      bool abort = true) {
+  if (code != cudaSuccess) {
+    throw cudaGetErrorString(code);
+  }
+}
+////////////////////////////////////////////////////////////////////////////////
+
+namespace polysolve
+{
+    LinearSolverCuSolverDN::LinearSolverCuSolverDN()
+    {
+        init();
+    }
+
+    void LinearSolverCuSolverDN::init()
+    {
+        cusolverDnCreate(&cuHandle);
+        cusolverDnCreateParams(&cuParams);
+        cudaStreamCreateWithFlags(&stream, cudaStreamNonBlocking);
+        cusolverDnSetStream(cuHandle, stream);
+    }
+
+    void LinearSolverCuSolverDN::setParameters(const json &params)
+    {
+        //TODO: make a json option for legacy solver vs new solver
```

Comment:
if it dosent have options, no need of this function

Reply:
i was going to include a choice for legacy vs. new solver but i can't see a use case for the legacy one